src/hooks/useLoadUserProfile.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { getUserProfile } from '@/services/users/getUserProfile';
+import { useAuthStore } from '@/stores/auth';
+import { mapUserProfileToAuthUser } from '@/utils/mappers';
+import { downloadImageService } from '@/services/images/downloadImage';
+import type { UserProfile } from '@/types';
+
+/**
+ * 사용자 프로필 조회 및 스토어 동기화 훅
+ * - TanStack Query의 캐싱 활용
+ * - 스토어와 자동 동기화
+ */
+export const useLoadUserProfile = () => {
+  const { setUser, setUserImageUrl, setIsLogin } = useAuthStore();
+
+  // 사용자 프로필 조회 (쿼리 중심, onSuccess로 스토어 동기화)
+  const profileQuery = useQuery({
+    queryKey: ['userProfile'],
+    queryFn: async (): Promise<UserProfile> => getUserProfile(),
+    staleTime: 5 * 60 * 1000, // 5분
+    gcTime: 10 * 60 * 1000, // 10분
+    enabled: false, // 필요 시 명시적으로 refetch
```

Comment:
```suggestion
    enabled: false, // 필요 시 명시적으로 refetch
    onSuccess: async (data) => {
      const authUser = mapUserProfileToAuthUser(data);
      setUser(authUser);
      // 이하생략....
```

아래에서 useEffect를 사용하는 대신, onSuccess 메소드를 활용하면 좀더 선언적으로 코드를 작성할 수 있을 것 같아요~!